.github/workflows/build.yml

```diff
@@ -0,0 +1,90 @@
+name: Build
+
+on:
+  push:
+    tags:
+      - '*.*.*'
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.x', 'pypy-3.7' ]
```

Comment:
When python `3.x` is defined here, do you know which python version exactly is targeted?

Reply:
it is latest version available for given distribution

Reply:
Asking the same question. I tried 3.7 with "runs-on: ubuntu-latest" and got error as "error::Version 3.7 with arch x64 not found%0AAvailable versions:%0A%0A".

Reply:
@yuyiguo what do you mean by saying "I tried 3.7", did you explicitly call this version, did you login to your docker image and verified that it is installed? Did you tried `python3`? There are so many possibilities here. But, if I use `act` to get `ubuntu` image, it provides me with `ghcr.io/catthehacker/ubuntu`, and if I login into it and run python I get this:
```
python
Python 3.8.10 (default, Sep 28 2021, 16:10:42)
```
So, as I wrote before, this is whatever latest 3.X is shipped with whatever image you use.

Reply:
In other words, the concrete python version really depends on image to be in use. The latest images uses the most recent python version, which is always updated when new latest image is available. In other words, if I run today latest image it would be version 3.X1, while in a month from now it may be 3.X2, etc. To have fixed version it is desired to specify concrete image tag.

Reply:
This is another example of keeping an environment as similar as the one used in production. Even an apparently minor difference of python 3.7 vs 3.8 version can cause and/or mask issues that we would only spot once the service is deployed in cmsweb.

In addition to that, we should have full control of the environment that we use for building this. "Latest" here can change at any time.

Reply:
@vkuznet 
jobs:
  build-n-publish:
    name: Build
    runs-on: ubuntu-latest
    steps:
    ...
    - name: Set up Python 3.x
      uses: actions/setup-python@v2
      with:
        python-version: 3.x
Now I got python 3.10 installed in my docker. My docker is running on my mac. but you have 3.8. So these are changing with different docker/config. Are there a place one can check for python version under the version of ubuntu , actions/setup-python and docker combination? I will the discussion to DBSClient CI/CD issues next.

Reply:
@yuyiguo I do not control images provided by upstream. I think everything new xxxx-latest image is created it has up to date software packages. You can use docker command itself and pull out any image, then you can login into image and check which python version it has.